packages/utils/src/promise.ts

```diff
@@ -120,18 +119,17 @@ export async function resolveOrRacePromises<T extends NonEmptyArray<PromiseLike<
 
     return promiseResults;
   } catch (err) {
+    if (err instanceof ErrorAborted) {
+      return promiseResults;
+    }
     if (err !== resolveTimeoutError) {
       throw err;
     }
   }
 
   try {
     await Promise.race([
-      Promise.any(
-        promiseResults
-          .filter((r) => (r as PromiseResult<unknown>).status === "pending")
-          .map((r) => (r as PromiseResult<unknown>).promise)
-      ),
+      Promise.any(promises),
```

Comment:
If one promise is already resolved in the upper scope, then `Promise.any` will quickly result and not try any of pending promises.

Reply:
Thats the intended behavior. First we try to wait for all promises for some time. Then we just pick whatever resolves first.

Example from our usecase:
- we wait for 2 seconds for both the builder and engine (if they both return early, then we return early)
- after that, we race the first successful result (if neither return, return after 12 seconds)